PIL/SpiderImagePlugin.py

```diff
@@ -174,8 +174,8 @@ def tell(self):
     def seek(self, frame):
         if self.istack == 0:
             raise EOFError("attempt to seek in a non-stack file")
-        if frame >= self._nimages:
-            raise EOFError("attempt to seek past end of file")
+        if not self._seek_check(frame):
+            return
```

Comment:
This was checking self._nimages, the common code checks self._n_frames, and this code isn't checked in the tests.